code/game/objects/items.dm

```diff
@@ -1227,3 +1227,6 @@ modules/mob/living/carbon/human/life.dm if you die, you will be zoomed out.
 
 /obj/item/proc/is_shovel()
 	return FALSE
+
+/obj/item/proc/gives_weather_protection()
+	return FALSE
```

Comment:
Imho this should include the weather it is protecting from as a parameter.
i.e. a umbrella might protect you during a "normal" rainshower, but when you get to a actual (snow-)storm the umbrella becomes useless.